gitbook/src/fundamentals/security.md

```diff
@@ -0,0 +1,57 @@
+---
+description: >-
+  Securing Your Erigon Node: Best Practices for RPC, Network, and Operational
+  Safety
+---
+
+# Security
+
+The security practices focus heavily on the RPC daemon since it's the primary external interface. The modular architecture allows you to run components separately, which can improve security by isolating services. For production deployments, consider using the distributed mode where services communicate via secured gRPC rather than running everything in a single process.
+
+## Network Security
+
+**Firewall Configuration**: Secure your network ports properly. The default ports include:
+
+* Port `30303` for P2P networking
+* Port `8545` for JSON-RPC (if enabled)
+* Port `9090` for private API communication
+
+**CORS Protection**: When exposing RPC endpoints, avoid using wildcard CORS domains. Instead, set specific hostnames or IP addresses to prevent cross-origin attacks.
+
+## RPC Security
+
+**Method Allowlisting**: Restrict available RPC methods using access control lists. Create a `rules.json` file specifying only the methods you need:
```

Comment:
here is how to use `rules.json`:  `rpcdaemon --rpc.accessList=rules.json` (or `erigon --rpc.accessList=rules.json`)